File: 1189 maximum-number-of-balloons/solution.cpp
class Solution {
public:
    int maxNumberOfBalloons(string text) {
        vector<int> c(26,0);
        int ans = 100;
        for(int i=0;i<text.size();i++) c[text[i]-'a']++;
        int b = c['b'-'a'];
        int a = c['a'-'a'];
        int l = c['l'-'a']/2;
        int o = c['o'-'a']/2;
        int n = c['n'-'a'];

         return min({b,a,l,o,n});
    }
};

File: 1189 maximum-number-of-balloons/solution.rs
use std::cmp::min;

impl Solution {
    pub fn max_number_of_balloons(text: String) -> i32 {
        let mut counts = [0; 26];
        for c in text.chars() {
            let index = (c as u8).wrapping_sub(b'a') as usize;
            if index < 26 {
                counts[index] += 1;
            }
        }
        let b = counts[(b'b' - b'a') as usize];
        let a = counts[(b'a' - b'a') as usize];
        let l = counts[(b'l' - b'a') as usize] / 2; 
        let o = counts[(b'o' - b'a') as usize] / 2;
        let n = counts[(b'n' - b'a') as usize];
        *[b, a, l, o, n].iter().min().unwrap_or(&0)
    }
}
